src/components/application_manager/src/commands/mobile/get_way_points_request.cc

```diff
@@ -32,18 +39,102 @@ void GetWayPointsRequest::Run() {
                  true);
 }
 
+bool GetWayPointsRequest::CheckResponseSyntax(
+    const smart_objects::SmartObject& obj) {
+  const char* str = obj.asCharArray();
+  const bool is_str_empty = str && !str[0];
+  return is_str_empty || CheckSyntax(str);
+}
+
+bool GetWayPointsRequest::ValidateSmartMapResponse(
+    const smart_objects::SmartObject& obj, ConstSetIter key) {
+  const smart_objects::SmartObject& value = obj[*key];
+  switch (value.getType()) {
+    case SmartType_Array:
+    case SmartType_Map: {
+      return ValidateResponseFromHMI(value);
+    }
+    case SmartType_String: {
+      return CheckResponseSyntax(value);
+    }
+    default:
+      // Catch SmartType_Integer or SmartType_Double
+      break;
+  }
+  return true;
+}
+
+bool GetWayPointsRequest::ValidateResponseFromHMI(
```

Comment:
@AGritsevich , in this case it will be harder to read. Actually on previous review, it was merged, and I needed to split this.